baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/EnforceLogSafeUseFormat.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.CompileTimeConstantExpressionMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.List;
+
+abstract class EnforceLogSafeUseFormat extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+    private final Matcher<ExpressionTree> methodMatcher;
+    private final Matcher<ExpressionTree> compileTimeConstExpressionMatcher =
+            new CompileTimeConstantExpressionMatcher();
+
+    EnforceLogSafeUseFormat(Matcher<ExpressionTree> methodMatcher) {
+        this.methodMatcher = methodMatcher;
+    }
+
+    protected abstract Description matchDescription(MethodInvocationTree tree);
+
+    @Override
+    public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        if (!methodMatcher.matches(tree, state)) {
+            return Description.NO_MATCH;
+        }
+
+        List<? extends ExpressionTree> args = tree.getArguments();
+        if (args.size() > 2) {
+            return Description.NO_MATCH;
+        }
+
+        if (args.size() == 2) {
+            ExpressionTree messageArg = args.get(1);
+            if (!compileTimeConstExpressionMatcher.matches(messageArg, state)) {
```

Comment:
This needs to check the type of args[1], otherwise we'll get false positives on `Objects.requireNonNull(obj, () -> calculateExpensiveMessage())` using the supplier overload.